sentry/src/lib.rs

```diff
@@ -1,22 +1,101 @@
-#[macro_use()]
-use slog::{crit, debug, info, o, Drain};
+#![deny(clippy::all)]
+#![deny(rust_2018_idioms)]
 
-pub Application<T: Adapter, S: Storage> {
-    // database to be intialised
-    storage: S,
-    adapter: T,
-    logger: slog::Logger,
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Error, Request, Response, Server, StatusCode};
+use primitives::adapter::Adapter;
+use primitives::Config;
+//use slog::Logger;
+
+pub mod routes {
+    pub mod channel;
+}
+
+pub struct Application<A: Adapter> {
+    adapter: A,
+    //    logger: slog::Logger,
+    clustered: bool,
+    port: u16,
+    config: Config,
+}
+
+impl<A: Adapter + Send + 'static> Application<A> {
+    pub fn new(
+        adapter: A,
+        config: Config,
+        /*logger: Logger,*/ clustered: bool,
+        port: u16,
+    ) -> Self {
+        Self {
+            adapter,
+            config,
+            //            logger,
+            clustered,
+            port,
+        }
+    }
+
+    pub async fn run(&self) {
+        let addr = ([127, 0, 0, 1], self.port).into();
+
+        dbg!(self.clustered);
+
+        let make_service = make_service_fn(move |_| {
```

Comment:
we only have 1 instance of the Application with a specified Adapter, the magic happens here, were it does create a service and this service is handling incoming connections, so it's cloning the `Adapter` & `Config` for each connection.